clients/src/main/java/org/apache/kafka/clients/producer/RoundRobinPartitioner.java

```diff
@@ -64,7 +64,11 @@ public int partition(String topic, Object key, byte[] keyBytes, Object value, by
     }
 
     private int nextValue(String topic) {
-        AtomicInteger counter = topicCounterMap.computeIfAbsent(topic, k -> new AtomicInteger(0));
+        AtomicInteger counter = topicCounterMap.get(topic);
+        if(counter == null) {
+            counter = new AtomicInteger(0);
+            AtomicInteger counter = topicCounterMap.putIfAbsent(topic, counter);
+        }
```

Comment:
currently the counter var is unused, this seems kind of an issue. Also the `counter` reference is kind of omni present having high coupling but low cohesion. This seems kind of bad taste.

<img width="610" height="151" alt="Image" src="https://github.com/user-attachments/assets/24ade5f7-ae0f-46c0-8c2f-b2794db64e58" />

```suggestion
        AtomicInteger counter = topicCounterMap.get(topic);
        if (counter == null) {
            return topicCounterMap.putIfAbsent(topic, new AtomicInteger(0)).getAndIncrement();
        }
```